### ISSUE TITLE
Skin Cleanup

### ISSUE THREAD

**Issue by mistervee** [mistervee/skin.madnox#18]
@bartolomesoriano @meowmoo @MadMikeDoc 

Hi all,

I am going to lock github access for a few weeks. RL has been a bit busy so I am falling behind with trying to keep this skin on track and am noticing numerous bugs. I need to review all the commits and reinstate media flags, finalize the all the textures, update any issues and clean up everything 100% before we start adding any thing new. If i dont do this my concern is the we might just end up back where we started. 

If you have any outstanding work you need to commit please push this to the repo in the next 12 hours.

**Comment by bartolomesoriano** [comment]
@mistervee, given that I am the noob here, I apologize for any issues I might have caused.  If you think it's best, take away my push rights to the repo and I will figure out how to submit a PR so that you can approve any code before it is added.  Again, I apologize for any inconvenience.

**Comment by mistervee** [comment]
@bartolomesoriano 
Everything is ok with your work and all of us as a team. No need for apology, your skills with skinning are great.

I have no intention of permanently taking away push rights. This is really a feature freeze so i can catch-up going through all the commits and finishing a lot of stuff off that i have only partially done.

**Comment by bartolomesoriano** [comment]
@mistervee thank you.

**Comment by mistervee** [comment]
@bartolomesoriano @meowmoo @MadMikeDoc 

A quick update.

This will take a bit longer than expected, i am working with Aries with the skin textures atm and polishing everything coded so far. I am hoping we will be 80-90% complete when this is all done (minus the library views and home screen widgets).

**Comment by bartolomesoriano** [comment]
Thanks @mistervee for the update.  Take your time, and thanks again for really digging into the code to make it as best as possible.

**Comment by mistervee** [comment]
These are the xml's that can be worked upon if anyone is up for it;

DialogPlayerProcessInfo.xml
DialogSlider.xml

DialogMusicInfo.xml
DialogVideoInfo.xml

MusicVisualisation.xml

script-RSS_Editor.xml
script-NextAired-TVGuide.xml
script-globalsearch-infodialog.xml
script-globalsearch-main.xml

script-cu-lrclyrics-main.xml

Note: Please do not add to variables, includes or any xml if you are updating. Only the ones listed. Variables and other xml updates can be added in future. This also includes textures, so please don't add any textures use the ones that are already on git.

**Comment by bartolomesoriano** [comment]
@mistervee, @meowmoo, @madmikedoc, I can take a stab at the first two, if everyone is OK with that.  The globalsearch ones look like Estuary, so those need the MadNox treatment.

**Comment by mistervee** [comment]
@bartolomesoriano 
Fine by me bart.

@bartolomesoriano @meowmoo @MadMikeDoc 
When this cleanup is complete, i would like to put this in the skin development forum WIP and then announce it in the madnox forum. I will make it a private repo as i don't want anyone copying and pasting textures or code (seems like a necessity as it has been done by various forum members). It does cost some $ to make it private, but not much to break the bank.

In regards to this statement for skins in the repo (http://kodi.wiki/view/Add-on_rules);
"No mods. Mods are great, and pop up frequently in the forums. But they are only confusing to non-forum-goers. This can be vague, each is evaluated on its own merit. By naming your "mod" different and make it substantial different enough, we will of course take it in consideration."

With my work with Aries this is no way an issue. It's a long journey (time frame unknown, hopefully for Leia Beta) to make all this happen but it will happen.

**Comment by MadMikeDoc** [comment]
Hi,

@bartolomesoriano @meowmoo @mistervee 

Looing forward to seeing the new textures etc. etc.

I'll take these 3 to start as I'm very familiar with all 3:

script-NextAired-TVGuide.xml
script-globalsearch-infodialog.xml
script-globalsearch-main.xml

But not adding vars etc. or textures could prove interesting but we'll see, have always wanted to re-design Next Aired and global search but will do basic versions 1st.

I may take lyrics as well as I recently did all the work adding manfeeds video lyrics into the Krpyton version but will see how I go with these 3 first.

Going on the WIP forum eh, unfortunately the forums have people who just rip and rip but let me know about costs for going private I'll contribute if possible

Mike.

**Comment by mistervee** [comment]
@MadMikeDoc 
Sounds good to me.

The WIP forums will be some time away. The cost is only $7 a month. So i am thinking at least a few months of having the repo private until an alpha version can be released.

**Comment by bartolomesoriano** [comment]
@mistervee, I am fine with a private forum for a while, and I can contribute for as long as we need it to be private.

**Comment by bartolomesoriano** [comment]
@mistervee, @MadMikeDoc, I know exactly how to test DialogPlayerProcessInfo, but I don't think I have seen DialogSlider.  Do you recall where or how we use it?

**Comment by mistervee** [comment]
@bartolomesoriano 
If you go into settings in game options, under keyboard. Changing the number of 'keyboard players' will bring up the slider.

**Comment by bartolomesoriano** [comment]
Thanks @mistervee!

**Comment by meowmoo** [comment]
good to read some news here :D
looking forward to the next big push

I could take a look @DialogMusicInfo.xml, DialogVideoInfo.xml, MusicVisualisation.xml
but i think its best to wait till @mistervee is done.

I will "clean up" MusicVisualisation.xml, so that there is only one layout, but we need to choose which one from the 4 we have in madnox, or something totally new?! (we can add them back later when everythig else is done)

**Comment by bartolomesoriano** [comment]
@meowmoo , I think you are the most knowledgeable of the team in terms of music functions and use.  My opinion is for you to choose the one you think is most useful, and start with that one.  

@MadMikeDoc , @mistervee , your thoughts?

**Comment by MadMikeDoc** [comment]
@meowmoo , yes this is a tricky one after working extensively with music in Aeon Madnox and with music viz by adding all the additional layout choices. I myself use what is called I think the 'seedyevil' layout but it it not that anymore after I completely re-wrote it not long ago and added a lot more to it but I know you use the one you update as well. Perhaps maybe go with the 1st is it the default Aeon Madnox music viz for the 1st basic introduction and like you said we'll add the others back in sooner or later?

I haven't seen anything 'new' to replace what we have but the problem is we a going from multiple choices to 1 to start with and probably just the default Madnox music viz for this?

**Comment by meowmoo** [comment]
yes im using the SiLVIO layout, because its simple and clean :)
but most useres wanna see lots of stuff cover/next track/codec/.... so Original and MooViz comes to mind, I never liked the Original one, that was the main reason i start looking at xml files ;P 
also MooViz already use Skinhelper, so its easier to "rewrite". we can still change the Layout, if someone done like it or come up with a good Mockup, i searched pinterest but cant find anything....

I have a busy weekend, so i cant start with MusicViz.xml anyway :/

**Comment by bartolomesoriano** [comment]
@mistervee, @MadMikeDoc, @meowmoo I pushed a first pass at DialogPlayerProcessInfo.  Please review and comment.  Thanks.

**Comment by MadMikeDoc** [comment]
global search 1st version is pretty much done but next aired is taking a bit more time as there is so much code to unravel. Will push both when ready.

**Comment by mistervee** [comment]
@MadMikeDoc 
All good mike take your time. I can imagine the next aired is a pain as it uses static content for the containers.

Just pushed a small update. I have added Variables_Temp.xml. If you need to add some please put them in there, as there is quite a bit of reshuffling i need to do.

NOTE: anything in vaiables.xml below the comment 'end rework' will be deleted/removed soon.


@bartolomesoriano @MadMikeDoc @meowmoo 

A quick update.

I needed to take a break for bit. So slowly now getting back into it. So in time i will be pushing some rather large updates. I will also post in this thread any additional xml's that can be updated while i am working on this if can think of any that is.

**Comment by mistervee** [comment]
@MadMikeDoc 
Another thought. There is a few commits that i would like to combine just for a nice OCD commit history.

So with your coding experience are you familiar with the gitshell and can you help out?

**Comment by bartolomesoriano** [comment]
@mistervee, I don't know if you are talking about the same thing, but with the Windows GitHub client I use you can pick a subset of modified files to commit together. Or are you referring to just some changes in the same files but not all?

**Comment by MadMikeDoc** [comment]
@mistervee that I think is beyond my gitshell knowledge I just push commit after commit. Perhaps Mr Silvo might be able to help? he seems to know a bit more on git usage?

**Comment by MadMikeDoc** [comment]
Hi,

@mistervee @bartolomesoriano @meowmoo 

Done 1st basics versions of global search and next aired. Gone with a unified visual approach for both of them.

Will now take lyrics and next up scripts and create new basic versions of them and add any required vars into Temp_variables.

Haven't pushed anything yet, will probably wait until after lyrics and next up are done.

Mike

**Comment by bartolomesoriano** [comment]
Great @MadMikeDoc ! Thanks.

**Comment by mistervee** [comment]
@MadMikeDoc 
Awesome. Looking forward to seeing the updates.

@bartolomesoriano @MadMikeDoc @meowmoo 
Hi all. 

It's been a while since any updates but i am still working on things. I have been feeling a bit unwell for a while so slow progress.

I have done quite a bit of work in all areas of the skin, so i figure i will push things in stages. I will try to keep the first post updated with my progress.

**Comment by bartolomesoriano** [comment]
Thanks @mistervee.  Hope you feel better soon.  I am really looking forward to all the updates; I have missed working on MadNox.

**Comment by meowmoo** [comment]
Looking forward to it, can't wait to see the new textures and improvements

**Comment by mistervee** [comment]
@bartolomesoriano @MadMikeDoc @meowmoo 

Quick question
Any thoughts on animated fanart?
Is this something that needs to stay in the skin?

**Comment by bartolomesoriano** [comment]
@mistervee , @MadMikeDoc , @meowmoo 

I have never used animated fanart.  Animated posters, yes, and I do like them.

@mistervee , when you say a global setting, you mean the ability to set one (or multiple) backgrounds for the skin as a whole instead of selecting a background for each menu option?

**Comment by mistervee** [comment]
I guess i didn't explain it to well.

For the skin backgrounds they will be individually customisable for each area of the skin.

The Animated fanart is not related to skin helper service, that will stay as it is. It's the kodi animation that make the background move around the screen.

So the option in the background chooser for aeon madnox for krypton.

**Comment by bartolomesoriano** [comment]
Thanks @mistervee.  No preference on my part, I have not used that option before.

**Comment by meowmoo** [comment]
I only use(d) it on MusicViz, to get some "dynamic", so from my part i dont need it

**Comment by mistervee** [comment]
Cheers for the response. I will just use animated as a global setting as it is easy to implement. So it will be in skin settings as opposed to customisable in each area of the skin.

@bartolomesoriano @MadMikeDoc @meowmoo 
Next question;
Artwork Cases - in krypton we have bdcase, case, clear case & glass
(i am working on the background picker and live backgrounds, like when you have a bunch of posters for recent media ect.)
https://github.com/mikesilvo164/skin.aeon.madnox/tree/krypton/media/homewidgets

These are used in many area's of the skin.

Clear case and glass will be the same. For the case and bdcase i would prefer not to use both versions of the poster type. It is to keep the skin size down and not to over complicate the coding.
1. https://github.com/mikesilvo164/skin.aeon.madnox/blob/krypton/media/homewidgets/bdcase/poster/1080p.png
2. https://github.com/mikesilvo164/skin.aeon.madnox/blob/krypton/media/homewidgets/case/poster/1080p.png


Really just needing some ideas and feedback on what to do here.

**Comment by MadMikeDoc** [comment]
@mistervee 

Hi, Sorry missed the animation q as not been well lately. For me I don't use that but I'm sure some users may so if we can look to keep it then ok, if not then no problem.

For case I use option 1 today, the bdcases. I've always preferred them to the other case options available. I know the solutions for all the case options can complicate and effectively double code etc.

But I know one of the last things Schimi did was the /banner & /landscape folders for /case and I like those and would like to keep them so from me it would be /bdcase with existing /poster & /square folders and then add the /banner & /landscape folders from /case to it and then that would be our 'color case' solution, thoughts?

Looking forward to your update, actually wanting to get back to the skin and starting moving forward on the 1st version.

**Comment by bartolomesoriano** [comment]
@mistervee, @MadMikeDoc, @meowmoo, 

I agree with Mike.  I also prefer the bdcase to the case type.  I was also thinking of how we can keep schimi's work on these, so I also agree with Mike's idea on what to keep in terms of our case solution.  

I was also wondering, and this is a question for you more skilled skinners, does it make sense to do resource image addons for this? Does it simplify the code any? I guess, if it is possible and it makes sense, that it helps in keeping the skin size down, and the addons can be set as dependencies if we want.

**Comment by meowmoo** [comment]
I dont use any Cases, so i leave this to the "CaseUsers"

1. or 4. would be my suggestion

**Comment by mistervee** [comment]
@bartolomesoriano @MadMikeDoc @meowmoo 
Cheers for your response. Seems like option 1 is the way to go. The banner/landscape cases wont change. Option 4 is possible but it means tweaking the case textures and extra coding. So i will scratch that one of the list. I was just concerned about kodi flix and 3D views and the extra textures needed which will increase the skin size a little to much. I suppose we can cross that bridge when we come to it and see how it works out.

@bartolomesoriano 
The resource addon works by setting skin strings and skin strings cannot be used in containers. So this has limited use. So am trying to avoid using it apart from the skin backgrounds.


**So I need some help on this part**
@bartolomesoriano @MadMikeDoc @meowmoo 
For the Widgets and live backgrounds. I am in still two minds what to do for user choice.

I am thinking along the line of only using playlists for most of them. The load quicker and and have many more list items available. Not sure on the refreshing part when a movie ect is watched. Should be simple as using the correct conditions within a smart playlists.

I need a volunteer to come up with bunch of playlists for, music, music videos, tv series, and movies and to test them out on krypton madnox or silvo to make sure they work after something is watched. Then i can incorporate them into the live backgrounds (in the future widgets).

**Comment by bartolomesoriano** [comment]
Thanks, @mistervee.  When you say using playlists for most of the widgets, I take it to mean those that do not need plugins like SHS or ExtendedInfo; for example, next aired episodes or upcoming movies, or trending movies;  am I looking at this correctly?

As for the playlists for movies, tv shows, episodes, music videos and music, I can come up with what I think are the most used ones.  Do you want me to test them as widgets in krypton madnox (or Leia Silvo), just to see if they get updated when an item is played or watched? Do you want me to test live backgrounds specifically as well? Please let me know, I can do that by the weekend.  Thanks.

**Comment by bartolomesoriano** [comment]
@mistervee  @MadMikeDoc  @meowmoo , how about the following playlists:
Movies: Recently added, Random, In-Progress, Unwatched, Best Picture, IMDb Top 250
TV Shows: Recently added, Random, In-Progress, Unwatched
Episodes: Recently Added, Random, In-Progress, Unwatched
Music Videos: Recently added, Random, In Progress, Unwatched, Most Watched
Albums: Recently added, Recently Played, Random, Unplayed, Most Played

Any that you think I should add?

**Comment by meowmoo** [comment]
@mistervee i tried a few playlist on Madnox 17 for example 
```
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<smartplaylist type="tvshows">
    <name>episoden neu laufende </name>
    <match>all</match>
    <rule field="playcount" operator="is">
        <value>0</value>
    </rule>
    <rule field="dateadded" operator="inthelast">
        <value>2 weeks</value>
    </rule>
</smartplaylist>

```
and the Widget updated after watching a unwatched episode
@all
i have done a basic MusicViz layout, not sure what info we wanna show in the default view, do we wannt Next/preview songs? do we need extra info like similar artist, artist bio, last.fm charts?
I have droped Artistslideshow support for now, because we use Skinhelper everywhere and it provide basically the same info.  (just artist bio is missing, at least its not in the wiki, need to ask marcel)
https://imgur.com/a/nlBFQ
hit me with feedback and stuff you wannt to have in the music viz :)

**Comment by mikesilvo164** [comment]
Hey guys :)

I tried playlists for widgets on SiLVO and they seem to load very slowly for me.  They do update if the playlist has the correct rule but it freezes Kodi when loading widgets on first scroll to menu item as well as when updating after watching...

Here is my tested PlayList:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<smartplaylist type="movies">
    <name>Random</name>
    <match>all</match>
    <rule field="playcount" operator="is">
        <value>0</value>
    </rule>
    <group>none</group>
    <limit>25</limit>
    <order direction="ascending">random</order>
</smartplaylist>
```

Not sure if it is particular to my setup but, not to brag, I have some fast hardware here...

I am not sure why it is slow for PlayLists. I'm pretty sure I remember noticing the same for Jarvis and Krypton but cannot be sure. I have not tested recently so my current findings are for Leia only.

Current PC Setup is:
1,902 movies
6,808 episodes
15,675 songs
NVMe SSD hitting target speeds of WRITE 3,400 MB/sec and READ 2,000 MB/sec per CrystalDiskMark
i7-7700k running at 4.55Ghz
32GB Ram

**Comment by mistervee** [comment]
@mikesilvo164 

Hi mike, thanks for your feedback.

mmm.... you have me thinking now? Random playlists would be an issue as they will update every time a menu item is focused or while navigating to a new window for live backgrounds. While something like skin helper widgets only update on certain conditions. 

I am wondering if all playlists update every time the content is called. If this is the case playlist's are not the ideal solution for widgets and backgrounds. Something i didn't think about when using playlists.

Skin helper widgets take 1s to update on skin load or refresh. This seems like it is due to dynamic content being loaded via the script. after that there is no delay.

I have similar specs to you mike, but half the library content.

But 3-5 secs seems like a long time. I have noticed in the past when the kodi db migrates between versions i see a slow down in some library views when scrolling (seems to depend on how the container.content is coded and the number of variables). A fresh data base has always fixed the issue in the past.

@all
Will have to rethink the playlist approach. The new madnox will be a faster but still will be a demanding skin.

**Comment by mikesilvo164** [comment]
@mistervee I have made some significant changes to the Widgets and template file recently.

If you do not mind. Can you please try using my latest master and adding a playlist as Widget to check your results?

I think that could confirm if it is an issue only on my end.

I'd hate to be responsible for slowing down improvements if what I experience is just because of some coding error I made :frowning:

**Comment by mistervee** [comment]
@meowmoo 
Looks good. my only thought would be the background texture. It would need to integrate nicely with the player controls with a slide animation. You could always change the back ground texture to a gradient or use the play control background texture.

If you were going to add anything extra i would use what ever skin helper provides.
https://github.com/marcelveldt/script.skin.helper.service/wiki/Listitem-Properties---Music-items

For the artist/album info SkinHelper.ListItem.Plot

**Comment by mistervee** [comment]
@mikesilvo164 
Sure thing. Downloading now.

**Comment by mistervee** [comment]
@mikesilvo164 
Just did some quick in skin tests with out looking at code. 

For movie menu using random movie playlist for Widget 1 & 2 & Live background 
For the playlist i had 1-2sec delay sometimes for the widget posters and instant other times (spamming refresh). For Every refresh or entering or exiting library to main menu the posters would be different.

For movie menu using random movie skin helper for Widget 1 & 2 then Live background playlist 
For Every refresh the widgets had a small delay in showing (likely skin help script delay but integer compare stop content showing until container was filled). Entering or exiting library to main menu the widgets had no delay after the skin has loaded. The live backgrounds were different each time, but had no delay.

With all this in mind my library atm is only 500 movies. I could imagine the delay being more with larger library's. 

For the new movie playlist no delay.

So my thoughts on all of this is playlists are slower than scripts for anything that uses a random method of querying the database.

**Comment by mistervee** [comment]
@bartolomesoriano 
With all this in mind i am going to use skin helper for all the live backgrounds and widgets. There are less info labels available but the performance of the skin will be slowed with playlists in certain situations.

**Comment by mikesilvo164** [comment]
@mistervee Been testing for a little bit and I only notice a freeze on first load of a new playlist Widget, first load of an updated playlist Widget, and lastly scrolling through a new playlist Widget. Did not notice anything when scrolling through an already scrolled through updated playlist but it was only one movie marked watched so that would seem to make no difference on my setup.

Reboots and skin reloads do not show any freezing for me once the new playlist Widget was loaded for that first time. I have definitely not tested to see any patterns over many days or weeks of library changes though.

Your 1-2 second delay seems like a slightly longer but close to normal expectation for me.

I tested meowmoo's playlist so for me it seems to have to do with the initial loading of the playlist Widget and not the random order aspect.

Not sure if any of the above excludes it for use but IMHO any freezing is not wanted, however, if it only freezes just the once it may be worthwhile if you compare cost to benefit.

Perhaps some more testing before chucking the idea out the window?

I do have a "feeling" that the playlist approach requires disk access from library item location more than scripts that may read from the DB but i am just guessing (talking out my bootyhole) on that assumption.

Does the rest of the MadNox crew have as large or larger libraries than my own to test to see if it is only on initial load of a playlist Widget regardless of random order or not...?

**Comment by mistervee** [comment]
@mikesilvo164 
Perhaps i missed that part of freezing. I do not have the skin freeze, just a delay in loading of the textures. I have had this issue in the past and a delete of the DB and re-scan fixed the issue for me. But that was only due updating the DB between versions of kodi. (Krypton/Jarvis/Leia)

@MadMikeDoc 
I am not not sure but i think mike has a big database of movies.

**Comment by mikesilvo164** [comment]
@mistervee That is one thing that I have not needed to do in a long while... I will not delete my DB unless absolutely necessary cause I have manually entered numerous changes like sort titles, available icons for studios, and all that stuff.

I do keep a clean textures DB using texture cache maintenance utility though about every month or two.

When your talking literally thousands of titles it becomes not worth it for me to delete all that just for testing purposes.

If there is something I can do or try please let me know how I can help in any way I could.
 😃

**Comment by mistervee** [comment]
@mikesilvo164 

Understandable. I am not a fan of it either. What i have done in the past is to install a second installation of kodi in a different directory and run it in portable mode. (make shortcut and add -p). Alternatively if you do not run in portable mode just make a shortcut with the suffix -p. Then you will have another folder in the kodi install folder 'portable_data'. If you run it from the shortcut all you settings are there and it preserves your previous settings.

**Comment by mikesilvo164** [comment]
Yes, I know that. I have a test version for Krypton and Jarvis installed with portable_data folders.

My reasoning is if the skin requires you to delete a DB file then it is a skin issue that should be fixed/resolved so the user does not have to delete everything to use said skin.

Does that make sense?

Just so I am clear... 
IMHO asking a user to try a portable version is fine but if they are on a platform other than Windows that does not allow that then they need to post a debug log so the skinner can check to see if it is addon, script, build, and/or possibly user error related.

I never recommend to delete your install to test if things worked unless that recommendation had to do with the user installing previously mentioned shitty addons, builds , or any other not recommended crap when they should be using Kodi the proper way.

Normal users should not have to delete anything just so a skin can work the way it should without touching any userdata.

I should be able to use a 6yr old DB file I have not upgraded in yrs with a skin compatible with the version of Kodi I am currently using. If that is not the case then it could be a Kodi bug that should be reported so the devs can fix.

I would never tell my users to delete any files so a skin will work (unless I suspect the things mentioned above 😉) cause a skin should be showing what is in the database so there should not be any issues affecting properly used INFOlabels at all.

I hope that makes some sense since I am running on like 2 hours of sleep and am about to crash for the night... 😏

**Comment by MadMikeDoc** [comment]
Hi,

My current setup is:
4496 movies
12566 episodes
75655  songs

Recently purchase a new HTPC:
Intel Core i3-6100 3.7GHz
8GB
120GB SSD
1TB HDD


With Krypton and generally using SHS widgets I get an initial wait upon start and then after that all widgets seem fine and responsive, but my Krypton setup seems to have some weird problems I just can't iron out, sometimes backgrounds stop showing or extrafanart stops working then widgets will stop refreshing and then every so often it seems the skin will just reload itself. All this is on Win10!.

but when it's working ok things seem fine with widget response after initial skin load.

@meowmoo 

That CD cover and disc just look to big from the picture but I guess it's best to set it in operation. I will be adding to MusicViz once updated as I prefer a different layout and will add that back in at some point.

**Comment by bartolomesoriano** [comment]
@mistervee , @mikesilvo164, @MadMikeDoc, @meowmoo 

I tried Mike Silvo's playlist with Silvo Leia under Windows 10 Enterprise.  My setup is a core i5, 16GB RAM, 240GB SSD, nVidia GTX 970, and all my media is on a Synology NAS hardwired through GB ethernet.  

I have a collection slightly larger than Mike Doc's (4700+ movies, 15k+ episodes, 226k songs) , and the widget loading for me is instant, as is the change once a movie has been watched.  Scrolling through the widget items is also smooth, and I have panel widget info turned on.

Do you want me to perform any other tests before making a final decision on using playlists for widgets or not? Please let me know.  Thanks.

**Comment by mistervee** [comment]
@mikesilvo164 
Fair point on the DB issues. My personal experience using pre alpha versions has sometime corrupted my database. Also have had issue when upgrading my router with different local ip's on my nas. I have nfo files and artwork stored along side my movies ect so my normal thing to do is recan. I have never tried to do any debugging on my end for suspected database issue. I will have to rethink this in the future.

The one thing i am sure of, with inefficient coding (to many stops and checks for kodi, too many variables string compares ect) dynamic database content can slow down. If you look at the list view for example; i coded the new version of madnox with one container and lots of variables and visible conditions and had a significant slow down while scrolling. Making multiple containers with one visible condition for each and limited variables solved the issue.

Cheers for you input. Makes a lot of sense.

Note: the 1-2 delay for me is due to a fresh setup and kodi having to cache the artwork.

@bartolomesoriano 
Lets go ahead with the playlists. They will be used for the live backgrounds and some of them for the widgets.

From your list i would go for the following with these alterations (only a suggestion so make them as you please, we can change them if required);
Movies: Recently added, Random, In-Progress, New, Sets, Recommended, IMDb Top 250
TV Shows: Recently added, Random, Recommended, New
Episodes: Recently Added, Random, Recommended, In-Progress, New
Music Videos: Recently added, Random, New, Most Watched
Artists: Random
Albums: Recently Added, Random, ,New, Recommended, Most Played
Songs: Recently Added, Random, ,New, Recommended, Most Played

Note:
- for recommend (videos); rule unwatched and a rating higher than x, random order
- for recommend (music); rule playcount and a rating higher than x, random order
- for New; order by air/release date.
- for recently added; order by date added
- item limit of 25

This is not a perfect list of ideas by any means. Something close to what i have suggested would be great. Feel free to change remove add whatever you think will or wont or work.

**Comment by mistervee** [comment]
@bartolomesoriano @MadMikeDoc @meowmoo 
``<variable name="ResCase1Var">
		<value condition="String.Contains(Container(9031).ListItem(0).Filename,4k)">4k.png</value>
		<value condition="Container(9031).ListItem(0).IsStereoscopic">3d.png</value>
		<value condition="String.Contains(Container(9031).ListItem(0).Filename,hdtv)">hdtv.png</value>
		<value condition="String.Contains(Container(9031).ListItem(0).Filename,webdl)">webdl.png</value>
		<value condition="String.IsEqual(Container(9031).ListItem(0).Label,Container(9031).ListItem(0).TVshowTitle)">tvseries.png</value>
		<value condition="!String.IsEmpty(Container(9031).ListItem(0).VideoResolution)">$INFO[Container(9031).ListItem(0).VideoResolution]p.png</value>
		<value condition="Container(9031).ListItem(0).IsCollection">set.png</value>
		<value>p.png</value>
	</variable>``

**Comment by bartolomesoriano** [comment]
@mistervee , @MadMikeDoc, @meowmoo , @mikesilvo164 

Answering the last question first, I don't think we need extra textures for SDTV and HDDVD.  However, I do think we need some additional string compares to not miss out on some user naming conventions.  One of the best compromises I have seen would be the following (I added some that I think are necessary to cover the widest possible audience)

`<value condition="[String.Contains(ListItem.Filename,4k) | String.Contains(ListItem.Filename,2160p) | String.Contains(ListItem.Filename,UHD) | String.Contains(ListItem.Filename,UltraHD) | String.Contains(ListItem.Filename,Ultra-HD)]">4k.png</value>
<value condition="ListItem.IsStereoscopic">3d.png</value>
<value condition="[String.Contains(ListItem.Filename,hdtv) | String.Contains(ListItem.Filename,pdtv) | String.Contains(ListItem.Filename,dsr)]">hdtv.png</value>
<value condition="[String.Contains(ListItem.Filename,webdl) | String.Contains(ListItem.Filename,web-dl) | String.Contains(ListItem.Filename,WEBRip)]">webdl.png</value>
<value condition="String.IsEqual(ListItem.Label,ListItem.TVshowTitle)">tvseries.png</value>
<value condition="!String.IsEmpty(ListItem.VideoResolution)">$INFO[ListItem.VideoResolution]p.png</value>
<value condition="ListItem.IsCollection">set.png</value>
<value>p.png</value>`

What do you think?

**Comment by bartolomesoriano** [comment]
@mistervee, in terms of the playlists, I like your list, and had a question on the recommended music; when you say rule playcount, do you mean that playcount = 0 or that playcount > 0?

Also, for the New playlists, the only type where there is an airdate is episodes; there is no release date for movies, tvshows, or music in the Kodi db, so I don't think I will add any of those.  Do you know of another way to get the "new" movies, etc. in a playlist?

**Comment by mistervee** [comment]
@bartolomesoriano 

I agree with covering a wide audience. Keep in mind the listitem.videoresolution covers everything apart from 3d movies, sets, tv shows. You can just have the corresponding file name to match the listitem.videoresolution to the case.

For 4k content, i would rather not use any string compares as kodi returns the video resolution as 4k. It would be best not to set any sort of file naming options as i don't see this has been done in the past.
 
The HDTV,SDTV,WEBTV personally is something i don't use, but seem to something that was used in various skins over the years. All my movies are simply moviename (year). For anything 3d i use this http://kodi.wiki/view/3D#Video_filenames_flags. Which works with the listitem.isstereoscopic.

My code preference would be;
``<value condition="ListItem.IsStereoscopic">3d.png</value>``
``<value condition="String.IsEqual(ListItem.Label,ListItem.TVshowTitle)">tvseries.png</value>``
``<value condition="!String.IsEmpty(ListItem.VideoResolution)">$INFO[ListItem.VideoResolution]p.png</value>``
``<value condition="ListItem.IsCollection">set.png</value>``
``<value>p.png</value>`` 

And may be considering changing some case colors.

It would be great to hear from @MadMikeDoc and @meowmoo  as they have large movie collections and might have further ideas.

**Comment by mistervee** [comment]
@bartolomesoriano 
Now that i think about it....  for recommended music would be what i listen to a lot, If the playcount was 0 i would look at what i personally rate it at and what the scraped rating is.  

For the 'new' seems like tv shows is a no brainer as it has always been a annoyance of mine. Like when i add a whole season of a show and the widget or live background shows the same cover. Then i don't know what the latest episode's are for that week. So instead of new call it recently aired.

For music, movies, tv series, music videos - I suppose you could go for a combination of year/date added and order by year. Not sure what i would call the playlist may be recent release?

**Comment by mistervee** [comment]
@bartolomesoriano @MadMikeDoc @meowmoo 
I must say i am taking longer than i thought with the aspects of the skin that i requested be put on hold. I am trying to perfect the base code and trying to get as much input as possible to help guide me along the way. I will get there soon enough and thanks for all the input, ideas and the coding you guys have done. So please bear with me on this slow journey.

**Comment by mikesilvo164** [comment]
I would like to mention that for TV Shows I currently use Skin Helper Recent TV Shows with "Group items  in recent episodes" option enabled because it consolidates entries instead of just showing the last whatever episode which could be the same show repeated again and again... For how I add and watch shows when usually adding whole seasons at a time it is problematic...

Scripts can take into account options that are irrelevant to the users library so playlists can and in my experience usually are the way to go. Not saying it works for all but for this example a playlist just will not do what I would like it too.

Food for thought? Maybe?

**Comment by mikesilvo164** [comment]
Concerning the VAR mentioned above...
I think adding an override based on filename is OK when considering certain circumstances. For HDDVD as an example... IMHO if the user has that in their filename one of two options are possible:

1st is the file is actually from a HDVD and the user would like that to be known
or
2nd The source of the file from the internet says it is HDDVD and the user might like that shown but possibly not

IMHO overriding by filename is OK if the usual filenames cannot be provided using INFOlabels.

In other words, if you have a 1080p file and without any override in the filename show BluRay by default cause that makes the most sense but... If I have an HDDVD from when they were available I should be able to display that if I want to add it to my filename...

All of the above that I mentioned is OFC IMHO 😃

**Comment by MadMikeDoc** [comment]
@mistervee 

Bit confused over this VAR change for cases especially considering what we do in Krypton seems to do the job:

<variable name="GlobalCaseVar">
		<value condition="[Container.Content(tvshows) | Container.Content(seasons)]">tvseries.png</value>
		<value condition="SubString(ListItem.Filename,3D) | SubString(ListItem.Filename,3d)">3d.png</value>
		<value condition="SubString(ListItem.Filename,.3D.) | SubString(ListItem.Filename,.3d.)">3d.png</value>
		<value condition="SubString(ListItem.Filename, 3D.) | SubString(ListItem.Filename, 3d.)">3d.png</value>
		<value condition="SubString(ListItem.Filename,.3DBD.) | SubString(ListItem.Filename,.3dbd.)">3d.png</value>
		<value condition="SubString(ListItem.Filename, 3DBD.) | SubString(ListItem.Filename, 3dbd.)">3d.png</value>
		<value condition="SubString(ListItem.Filename,.HDDVD.) | SubString(ListItem.Filename,.hddvd.)">hddvd.png</value>
		<value condition="SubString(ListItem.Filename, HDDVD.) | SubString(ListItem.Filename, hddvd.)">hddvd.png</value>
		<value condition="SubString(ListItem.Filename,.SDTV.) | SubString(ListItem.Filename,.sdtv.)">sdtv.png</value>
		<value condition="SubString(ListItem.Filename, SDTV.) | SubString(ListItem.Filename, sdtv.)">sdtv.png</value>
		<value condition="SubString(ListItem.Filename,.HDTV.) | SubString(ListItem.Filename,.hdtv.)">hdtv.png</value>
		<value condition="SubString(ListItem.Filename, HDTV.) | SubString(ListItem.Filename, hdtv.)">hdtv.png</value>
		<value condition="SubString(ListItem.Filename,WEBDL) | SubString(ListItem.Filename,webdl)">webdl.png</value>
		<value condition="SubString(ListItem.Filename,WEBDL-) | SubString(ListItem.Filename,webdl-)">webdl.png</value>
		<value condition="SubString(ListItem.Filename,WEB.DL) | SubString(ListItem.Filename,web.dl)">webdl.png</value>
		<value condition="SubString(ListItem.Filename,.WEBDL.) | SubString(ListItem.Filename,.webdl.)">webdl.png</value>
		<value condition="SubString(ListItem.Filename, WEBDL.) | SubString(ListItem.Filename, webdl.)">webdl.png</value>
		<value condition="SubString(ListItem.Filename,.WEB-DL.) | SubString(ListItem.Filename,.web-dl.)">webdl.png</value>
		<value condition="SubString(ListItem.Filename, WEB-DL.) | SubString(ListItem.Filename, web-dl.)">webdl.png</value>
		<value condition="SubString(ListItem.Filename,.4K.) | SubString(ListItem.Filename,.4k.)">4k.png</value>
		<value condition="SubString(ListItem.Filename, 4K.) | SubString(ListItem.Filename, 4k.)">4k.png</value>
		<value>$INFO[ListItem.VideoResolution]p.png</value>
	</variable>

**Comment by mikesilvo164** [comment]
I really do not want to step on anyone's toes here so, I will respond to @MadMikeDoc and be gone if any of you feel like I am being a "doof".

Prerequisite IMHO:

To me @mistervee wants to streamline things, including VAR's, which is the way to go in my opinion so dropping filename conventions for 3d with any file Jarvis and above should be the way forward.

Reasoning is simple due to Kodi 3D playback being based on what the database knows and it gets that from streamdetails when added, filename if applicable, and/or manual selection using the video settings dialog if all else fails. None of that is skin dependent so all the extra 3D stuff can likely go bye bye.

Anywhoo I do not want to poke my nose where it does not belong so I will say the above and please let me know if you guys think I can add to the discussion 😃

**Comment by bartolomesoriano** [comment]
@mikesilvo164, IMHO you always add value to the discussion, so thanks!

@mistervee, for my own use, I agree with your take on the VAR, since I don't even use 3D, and all my movies and music videos and TV shows and episodes always have a video resolution present (I scrape with Emby server and complete any tagging with Media Companion).

However, I am concerned with more general use; does every Kodi scraper scrape a video resolution into the db? If not, then most likely we would be showing an incorrect case until the user plays one of the items (Kodi will set the correct resolution after playback).  What do you think?

**Comment by MadMikeDoc** [comment]
Gotcha, no need to worry @mikesilvo164, always welcome mate.

**Comment by meowmoo** [comment]
I also think we dont need SDTV oder HDDVD, HDDVD was alive 2 years, if someone really own a HDDVD its his fault :P 
I also just use moviename (year). as filename, and dont use any cases so its hard for me to judge here, but like @mikesilvo164  says @mistervee whant to make it as "clean" as possible, and i'm all in for it. There will always be a user that use "web-dl" and not webdl, but this one user could add this extra string by himself or rename the file.

@bartolomesoriano when I add a new movie, and focus it, it take a few sec and then the (right)resolution icon shows up, so Kodi is checking the Metadata without me starting the Movie.

@mistervee take all the time you need :)

**Comment by bartolomesoriano** [comment]
Thanks @meowmoo! Given that, I am also all for the streamlining of the case VAR.

**Comment by mistervee** [comment]
@mikesilvo164 
Cheers mike. You explained it clearly. Spot on. 

Might be an aussie thing as my explanations are not always on par.

@bartolomesoriano @MadMikeDoc @meowmoo 
Seems like for @bartolomesoriano and @meowmoo it will have no impact on your DB/Filenames.

For @MadMikeDoc i am not sure if you like to use file names for cases like hdtv sdtv ect.
I am not for covering the 1% user base that use file name's to show cases. Streamlining is the way to go.

So my question for @MadMikeDoc is do you use HDTV, SDTV, WEBDL ect for your tvshow, movie library in the filename?

This will have an impact on coding. It is team effort and i want to cover only what Team Madnox uses.

**Comment by mistervee** [comment]
@bartolomesoriano 
The playlists would be great when you can get get them done.

**Comment by MadMikeDoc** [comment]
@mistervee 

I don't use those types, maybe 1 or 2 somehow but it won't be a problem. I have named my files with dvd or Bluray or 3D but resolution should sort that out.

**Comment by bartolomesoriano** [comment]
@mistervee, real life has been getting in the way of Kodi time for me lately.  I will create and upload to \extras\playlists between today and tomorrow.

**Comment by bartolomesoriano** [comment]
@mistervee, I have pushed the playlists.  There is no way I could find to make a recently added songs playlist, just like I could find no way to make a recently added albums playlist.  This last one is not an issue, since there is a Recently Added Albums node in the music library.  In terms of New, except for episodes, I opted for Recently Released for movies, tv shows and music videos.  This relies on Year, which seems to be fine for movies and tv shows.  I am not sure for music videos; it seemed to be fine in my tests.

For Recommended, for albums and songs,  I used a Rating greater than 7, but I could not test it.  Rating is not one of the tags that I use with my music.  I decided to create 2 additional playlists, My Recommended Albums and My Recommended Songs, based on userrating.  I selected also My Rating greater than 7 for those.

Please let me know if these are OK, and if you would like me to add or change anything.  Thanks.

**Comment by MadMikeDoc** [comment]
@mistervee 

Have I missed something here? what is the purpose of all these playlists? Are widgets going to be playlist only due to 'speed'? Where speed is completely controlled by users hardware? Will the existing library data provider / SHS options etc. still be available for widgets?

@bartolomesoriano 

Those Recently released playlists don't sound correct, although it may work it is not 'recently added'? what happens if I buy an old tv series, which I do all the time, and rip it? it won't be in the widget due to it having an old 'year'will it?

Also a recently added albums should just be 'date added' descending. I'm sure I've created that in the past, will take a look!

**Comment by bartolomesoriano** [comment]
@madmikedoc, the recently released are just that, what has most recently been premiered or released.  There is no equivalent, for movies. Tv shows and music videos, in the Kodi db, to the air date that episodes have.  Thus, I am using year, which, from the tests I did, takes the complete date a movie or tv show was premiered.

Recently added are separate playlists, you will find I created them as well.  In Leia, I could not find date added as a sort by field for albums in the GUI.  I did not attempt to put it manually in the xsp file.

**Comment by mistervee** [comment]
@bartolomesoriano 
Thanks for the playlist. RL comes first so take your time what ever you do. 

What you have done is great. Things can be tweaked in future if needed. I really just needed these for the live backgrounds so i can finish off that part.

@MadMikeDoc 
The playlist are for the widgets and live backgrounds. I am currently working on live backgrounds.
Playlist generally load faster apart from random ones, they also provide more info labels.

When we get to the widgets i imagine it will be a combo of both. Preference is using a playlist and SHS will be used when a playlist cant do what is required. So at this point i am not 100% sure with the widgets how it will work out. They will be mostly re-coded.

**Comment by mistervee** [comment]
@MadMikeDoc @meowmoo @bartolomesoriano 

A quick question in regards to textures, playlists and skin files in general.
Is there any limitations on file naming for the various operating systems?

I have been using no spaces, no capital letters. Just lowercase and a '-' (dash) in between.

For xml's using cap's and no spaces.

I have in the back of my mind @MadMikeDoc there was some file naming issue with textures not working in some cases.

**Comment by MadMikeDoc** [comment]
@mistervee 

one system, andrioid I think, couldn't handle uppercase in textures so everything is lower case. Think that was the only issue of this kind and Windows is no problem as it doesn't care. Don't thiml xml names are a problem just anything in /media and it's sub-dirs.

How's it going?

**Comment by mistervee** [comment]
@MadMikeDoc 

Thanks for the info. I was not sure what platform so i will do as you have suggested.

Going ok thanks. The skin progress is slow and progressing. I am wanting to push a commit but still trying to hammer down the foundation. Madnox is such a feature rich skin and perfection is my a goal.

**Comment by meowmoo** [comment]
the only thing i know is linux is case sensitive and windows dont care :)
